web3/_utils/module_testing/eth_module.py

```diff
@@ -196,6 +196,23 @@ def test_eth_sign(self, web3, unlocked_account_dual_type):
         )
         assert new_signature != signature
 
+    def test_eth_signTransaction(self, web3, unlocked_account, geth_signed_tx=None):
+        txn_params = {
+            'from': unlocked_account,
+            'to': unlocked_account,
+            'value': 1,
+            'gas': 21000,
+            'gasPrice': web3.eth.gasPrice,
+            'nonce': 0,
+        }
+        COINBASE_PK = '0x58d23b55bc9cdce1f18c2500f40ff4ab7245df9a89505e9b1fa4851f623d241d'
+        result = web3.eth.signTransaction(txn_params)
+        actual = web3.eth.account.signTransaction(txn_params, COINBASE_PK)
+        if geth_signed_tx:
+            assert result['raw'] == geth_signed_tx
+        else:
+            assert result['raw'] == actual.rawTransaction
```

Comment:
@carver Any thoughts on a better test for this? I'm having trouble figuring out why the `eth_signTransaction` call to geth is behaving differently from parity - any possible insight into that?

Also according to the [spec](https://github.com/bitpshr/EIPs/blob/147609158191a701cf02c8556d5e529ffcb31cd2/EIPS/eip-1474.md) the jsonrpc response should only include the raw signed transaction. However, both parity and geth also return the original tx params submitted in the request. What behavior do we want `web3.eth.signTransaction` to emulate? Just simple return the signed tx? or also include the original transaction params.

Reply:
I don't think it is Web3.py's job to enforce spec compliance, that is going to be a separate thing that clients start testing for.  For now, we should probably be quite forgiving with what data is returned (with respect to allowing these extra fields and formatting them in the expected way)

In the future, I could see it being reasonable for web3.py to even decode the transaction and populate these fields when the clients stop returning them.

Reply:
Agree with :arrow_up: